executor/android/android_seccomp.h

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <linux/audit.h>
+#include <linux/seccomp.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <sys/prctl.h>
+#include <sys/syscall.h>
+
+#include <vector>
```

Comment:
I think we get away with something like:
```
struct Filter {
  static const kMaxSize = ...;
  sock_filter data[kMaxSize];
  size_t count = 0;
  void push(sock_filter v)
  {
    if (count == kMaxSize) fail(...);
    data[count++] = v;
  };
};
```

We get this far without linking in C++ stdilb and without any dynamic memory allocations. Dynamic memory allocations may produce unexpected syscall patterns. And linking in stdlib makes me nervous, there sure something will break on all different linux distros, or C++ compilation on some *BSD, Fuchsia, Akaros may also break (does Akaros even have C++ compiler?). And we sure should not ever show any C++ in repros to upstream devs, or we will be teared apart :)

I would prefer 5 additional lines of simple code rather than pulling in c++ lib and changing compilation mode.